File: src/main/java/com/example/demo/models/Company.java
/*-----Company-----
 * C'est la marque des armes et des munitions
 */
package com.example.demo.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
public class Company {


    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;
    private String name;
    private String year;
    private String country; 
    private String description;
    private String logo;
    
    public Company(String name, String year, String country, String description, String logo){
        this.name = name;
        this.year = year;
        this.country = country;
        this.description = description;
        this.logo = logo;
    }
}

File: online_armory/frontend/src/app/company/company.ts
import { Country } from "../country";

export class Company {
    id: number;
    name: string;
    year: number;
    country: Country;
    description: string;
    logo: string;

    constructor(
        name: string = "",
        year: number = 0,
        country: Country = null,
        description: string = "",
        logo: string = "",
    ){
        this.name = name;
        this.year = year;
        this.country = country;
        this.description = description;
        this.logo = logo;
    }

}